src/main/java/io/pinecone/clients/Pinecone.java

```diff
@@ -1034,13 +1046,23 @@ public Builder withProxy(String proxyHost, int proxyPort) {
          */
         public Pinecone build() {
             PineconeConfig config = new PineconeConfig(apiKey, sourceTag, proxyConfig, customOkHttpClient);
+            config.setTLSEnabled(enableTls);
```

Comment:
So this is only needed for `PineconeConfig` and we don't need it for the `ApiClient` itself?

Reply:
Yes, this enableTls flag is only needed for gRPC connection so the pineconeConfig object is set and further fetched at the time of establishing gRPC connection.